File: src/main/java/me/arasple/mc/traction/acts/ActionClose.java
package me.arasple.mc.traction.acts;

import me.arasple.mc.traction.base.AbstractAction;
import org.bukkit.entity.Player;

/**
 * @author Arasple
 * @date 2019/12/28 19:07
 */
public class ActionClose extends AbstractAction {
    @Override
    public String getName() {
        return "close|shut";
    }

    @Override
    public void onExecute(Player player) {
        player.closeInventory();
    }
}

File: src/main/kotlin/me/arasple/mc/trmenu/api/action/impl/ActionClose.kt
package me.arasple.mc.trmenu.api.action.impl

import me.arasple.mc.trmenu.api.action.base.AbstractAction
import me.arasple.mc.trmenu.api.action.base.ActionOption
import org.bukkit.entity.Player

/**
 * @author Arasple
 * @date 2021/1/29 22:25
 */
class ActionClose : AbstractAction() {

    override fun onExecute(player: Player, placeholderPlayer: Player) {
        player.getSession().close(closePacket = true, updateInventory = true)
    }

    companion object {

        private val name = "close|shut".toRegex()

        private val parser: (Any, ActionOption) -> AbstractAction = { _, _ ->
            ActionClose()
        }

        val registery = name to parser

    }

}